internal/controllers/common_controller.go

```diff
@@ -117,3 +106,36 @@ func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message strin
 		})
 	}
 }
+
+// setProgressingStatusConditionSuccess sets the progressing status condition to false for a successful install or upgrade.
+func setProgressingStatusConditionSuccess(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeProgressing,
+		Status:             metav1.ConditionFalse,
+		Reason:             ocv1alpha1.ReasonSuccess,
```

Comment:
Nit: something like `ReasonReachedDesiredIntent` seems more apt for the `False` status of the progressing condition. WDYT?